doc/specs/#16599 - Quick Fix.md

```diff
@@ -0,0 +1,58 @@
+---
+author: Carlos Zamora @carlos-zamora
+created on: 2024-04-01
+last updated: 2024-04-03
+issue id: "#16599"
+---
+# Quick Fix
+## Solution Design
+Quick Fix will be a new UI surface that allows the user to interact with the terminal and leverage the context of a specific command being executed. The UI is discussed in the [UI/UX Design section](#ui/ux-design).
+
+The UI and any logic populating the UI will entirely live in the TerminalControl project (similar to the context menu). The suggestions will be appended as top-level items, but a nested design can be revisited as more sources are added to populate the Quick Fix menu. Only one Quick Fix menu will be available at a time as it only appears on the latest prompt if an error occurred; this is in alignment with VS Code's design.
```

Comment:
What if shell integration isn't enabled? We don't confidently know where prompts are (like they do). 

IMO we should display the icons in the row that requested the suggestion (i.e. on the `'foo' is not recognized as an internal or external command,` line). We can always prune that to "the most recent line with suggestions" or "if shell integration is enabled, all suggestions since the last prompt" if needed

Reply:
Funny enough, CMD doesn't have shell integration and that's the only spot this VT sequence is being used (currently).

Discussed in spec review meeting. The cost of storing the metadata in the rows is relatively high when compared to how often something like this would be useful. Especially since we're only showing one quick fix menu at a time. We're choosing to go with using the cursor position to figure out where to draw the quick fix menu and avoid storing the metadata, so this won't be possible under that architecture (at least not immediately easily).

If this is something we want later, we'll revisit it.